libs/sdk-core/src/lnurl/auth.rs

```diff
@@ -0,0 +1,132 @@
+use crate::{mnemonic_to_seed, LnUrlAuthRequestData, LnUrlWithdrawCallbackStatus};
+use anyhow::{anyhow, Result};
+use bitcoin::secp256k1::{Message, Secp256k1};
+use bitcoin::util::bip32::{DerivationPath, ExtendedPrivKey};
+use bitcoin::{KeyPair, Network};
+use bitcoin_hashes::hex::ToHex;
+use bitcoin_hashes::{sha256, Hash, HashEngine, Hmac, HmacEngine};
+use reqwest::Url;
+use std::str::FromStr;
+
+use LnUrlWithdrawCallbackStatus as LnUrlAuthCallbackStatus;
+
+const TWO_POW_31: u32 = 2_u32.pow(31);
+
+/// Performs the second and last step of LNURL-auth, as per
+/// <https://github.com/lnurl/luds/blob/luds/04.md>
+///
+/// See the [parse] docs for more detail on the full workflow.
+pub(crate) async fn perform_lnurl_auth(
+    network: Network,
+    req_data: LnUrlAuthRequestData,
+) -> Result<LnUrlAuthCallbackStatus> {
+    // TODO Clarify how we get the seed. As arg?
+    let master_prv_key = mnemonic_to_seed("abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about".to_string())?;
```

Comment:
BreezServices::init_services is passed with the seed. We can derive the key we need at that point and save it as a field in BreezServices.